src/ut/sproutconnection_test.cpp

```diff
@@ -0,0 +1,155 @@
+
+    // If we don't override the default behaviour, return a nonsensical HTTP Code
+    ON_CALL(*_mock_http_client, send_request(_)).WillByDefault(Return(HttpResponse(-1, "", {})));
+  }
+
+  virtual ~SproutConnectionTest()
+  {
+    Mock::VerifyAndClear(_mock_http_client);
+
+    delete _mock_http_client; _mock_http_client = nullptr;
+
+    // We don't delete the HttpConnection as the SproutConnection does that for us
+    delete _sprout_conn; _sprout_conn = nullptr;
+  }
+
+  MockHttpClient* _mock_http_client;
+  SproutConnection* _sprout_conn;
+};
+
+const std::string SproutConnectionTest::IMPU = "sip:impu@example.com";
+const std::string SproutConnectionTest::IMPI = "_impi@example.com";
+const std::vector<std::string> SproutConnectionTest::IMPIS = { "_impi1@example.com", "_impi2@example.com" };
+const std::vector<std::string> SproutConnectionTest::IMPUS = { "sip:impu1@example.com", "sip:impu2@example.com" };
+const std::string SproutConnectionTest::IMS_SUBSCRIPTION = "<?xml version=\"1.0\"?><IMSSubscription><PrivateID>" + IMPI + "</PrivateID><ServiceProfile><PublicIdentity><Identity>" + IMPU + "</Identity></PublicIdentity></ServiceProfile></IMSSubscription>";
+const std::string SproutConnectionTest::dereg_body = "{\"registrations\":[{\"primary-impu\":\"sip:impu1@example.com\",\"impi\":\"_impi1@example.com\"},{\"primary-impu\":\"sip:impu1@example.com\",\"impi\":\"_impi2@example.com\"},{\"primary-impu\":\"sip:impu2@example.com\",\"impi\":\"_impi1@example.com\"},{\"primary-impu\":\"sip:impu2@example.com\",\"impi\":\"_impi2@example.com\"}]}";
+const std::string SproutConnectionTest::dereg_body_no_impis = "{\"registrations\":[{\"primary-impu\":\"sip:impu1@example.com\"},{\"primary-impu\":\"sip:impu2@example.com\"}]}";
+const std::string SproutConnectionTest::change_ids_body = "{\"user-data-xml\":\"<?xml version=\\\"1.0\\\"?><IMSSubscription><PrivateID>_impi@example.com</PrivateID><ServiceProfile><PublicIdentity><Identity>sip:impu@example.com</Identity></PublicIdentity></ServiceProfile></IMSSubscription>\"}";
+
+
+TEST_F(SproutConnectionTest, DeregisterBindingsWithNotifications)
+{
+  // Create a response that will be returned
+  HttpResponse resp(HTTP_OK, "", {});
+
+  // Expect that the request is sent, and set it to return the response
+  EXPECT_CALL(*_mock_http_client, send_request(AllOf(IsDelete(),
+                                                     HasBody(dereg_body),
+                                                     HasTrail(FAKE_TRAIL_ID),
+                                                     HasPath("/registrations?send-notifications=true"))))
+    .WillOnce(Return(resp));
+
+  // Actually deregister_bindings
+  HTTPCode result = _sprout_conn->deregister_bindings(true, IMPUS, IMPIS, FAKE_TRAIL_ID);
+
+  // Expect that we get the correct return code
+  EXPECT_EQ(HTTP_OK, result);
+}
+
+TEST_F(SproutConnectionTest, DeregisterBindingsWithoutNotifications)
+{
+  // Create a response that will be returned. This one will return an error
```

Comment:
"this one will return an error": returns HTTP_OK
Feels like one of these lines is out of sync